### PR TITLE
Add blog post on Kùzu and graph data

### PR DESCRIPTION
**Pull request by d33bs**
This PR adds a blog post covering Kùzu, a graph database and client which might be useful to research developers. Along with this content I tried to focus on covering how tabular and graph data are useful side-by-side (as I feel they often are pitted against one another as a false dichotomy).

Thanks for any thoughts or feedback you may have!

### CODE REVIEW

**Review by vincerubinetti — APPROVED** [review]
Nicely written post.

**Review by the-mayer — APPROVED** [review]
As someone who is "simply curious about the benefits of graph databases" I found this post very informative.

**Review by falquaddoomi — APPROVED** [review]

File: _posts/2024-05-24-Leveraging-Kùzu-and-Cypher-for-Advanced-Data-Analysis.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+It supports a wide range of operations, including pattern matching, filtering, aggregation, and graph traversal, enabling efficient exploration and manipulation of graph structures.
+For example, a basic Cypher query to find all nodes connected by a "FRIEND" relationship might look like this: `MATCH (a)-[:FRIEND]->(b) RETURN a, b`, which finds and returns pairs of nodes a and b where a is connected to b by a "FRIEND" relationship.
+
+## Kùzu
+
+{% include figure.html image="images/kuzu_intro.png" caption="Kùzu provides a database format and query engine accessible through Python and other languages by using Cypher queries." %}
+
+[Kùzu](https://github.com/kuzudb/kuzu) is an embedded graph database and query engine designed to integrate seamlessly with Python, Rust, Node, C/C++, or Java software.
+Kùzu is optimized for high performance and can handle complex graph queries with ease.
+Querying graphs in Kùzu is performed through Cypher, providing transferrability of queries in multiple programming languages.
```
[the remark this answers is not in the document]
Comment (d33bs) [reply _posts/2024-05-24-Leveraging-Kùzu-and-Cypher-for-Advanced-Data-Analysis.md:96]: Thanks, great point, I'll revise as you suggested!

**Comment by d33bs** [comment]
Thanks @vincerubinetti, @the-mayer, and @falquaddoomi for your reviews! Merging this in.

**Pull request merged**